### DISCUSSION TITLE
QoL - Make inline LaTeX slightly easier to use

### DISCUSSION THREAD

**Discussion by ir0nstone** [GitbookIO/community#111] [Feature Requests]
### What's your feature/idea?

Currently, if you want to write inline maths, you use `$$` before and after. This works perfectly, but there is not (to my knowledge!) any way to break out of this mode easily. The only way to break out is either to hit **Enter** or right-click, and both options feel very clunky. **Enter** causes a newline to be created as well, which isn't really desired, while right-clicking requires moving the hand from the keyboard to a mouse/trackpad. Neither of these is desirable when writing inline maths a great number of times, as they break the typing flow; it's quite painful to have to hit backspace after every bit of maths.

This is opposed to something like Notion, where I started writing my maths notes despite the worse access for others simply because closing the `$$` is automatic and hitting **Enter** after writing inline maths just brings you to the end of the block, meaning the typing speed when writing notes is just massively increased.

It's very possible that this feature does indeed exist somewhere in GitBook, and I simply do not know, and I apologise if this is the case!

### What's your desired outcome?

I humbly suggest two changes I think would help massively:

1 - Hitting **Enter** in inline math mode (when you are editing it in the black dropdown) should only leave the edit mode, not insert a newline as well. I don't see any use case where you are writing inline maths and _want_ a newline after it; you'd either write a **Block Equation** or probably want words/punctuation afterwards. The cursor should be placed right after.

2 - Closing the `$$` shouldn't put you in inline edit mode, it should render the LaTeX and let you continue typing normally. The chances of making an issue in inline mode are probably quite low (especially if you are just rendering variables such as `v` or sets such as `\N` by themselves) and in those cases, it's simple to just hit the left arrow and edit it; otherwise, you have to hit **Enter** immediately after closing every `$$`, which feels clunky.

### What's the impact of this for you?

- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

**Comment by PlethoraChutney** [comment]
Agree with all of this, but would especially love to have request 1 implemented!
